site/content/kapp-controller/docs/develop/kctrl-package-authoring-from-source.md

```diff
@@ -0,0 +1,184 @@
+More details about `kbld` configuration can be found [here](/kbld/docs/latest/config/).
+
+Great! We now have our configuration in place.
+
+## Setting up `kctrl` 
+We can get `kctrl` up and running using the `init` command.
+```bash
+$ kctrl package init
+```
+
+This command asks a few questions regarding how we want to build our package.
+In this case, we want to build a package `simple-app.carvel.dev` from the local directory.
+The configuration for the same can be found in the `config` directory. The interaction looks something like this,
+
+![Package Init for Simple App](/images/kctrl/pkg-init-simple-app.png)
+
+This command generates some files that tell `kctrl` how it should put the package together!
+
+## Releasing the package
+
+We can release the first version of our carvel package using the `release` command now!
+```bash
+$ kctrl package release --version 1.0.0
+```
+At this step we need to tell kctrl where we want to push our `imgpkg` bundle.
+Which is essentially an OCI image that contains all necessary config.
+The interaction looks something like this,
+
+![Package Release for Simple App](/images/kctrl/pkg-release-simple-app.png)
+
+This command creates the Package and PackageMetadata resources in directory `carvel-artifacts`.
+```yaml
+# carvel-artifacts/packages/simple-app.carvel.dev/package.yml
+apiVersion: data.packaging.carvel.dev/v1alpha1
+kind: Package
+metadata:
+  name: simple-app.carvel.dev.1.0.0
+spec:
+  refName: simple-app.carvel.dev
+  releasedAt: "2022-08-08T20:46:18Z"
+  template:
+    spec:
+      deploy:
+      - kapp: {}
+      fetch:
+      - imgpkgBundle:
+          image: index.docker.io/100mik/simple-package@sha256:dbf26c20859b32c0e08711c3af28844cc3e54968c3fa39e1975912ccbbb52899
+      template:
+      - ytt:
+          paths:
+          - config
+      - kbld:
+          paths:
+          - '-'
+          - .imgpkg/images.yml
+  valuesSchema:
+    openAPIv3:
+      default: null
+      nullable: true
+  version: 1.0.0
+# carvel-artifacts/packages/simple-app.carvel.dev/metadata.yml
+apiVersion: data.packaging.carvel.dev/v1alpha1
+kind: PackageMetadata
+metadata:
+  name: simple-app.carvel.dev
+spec:
+  displayName: simple-app
+# longDescription: Detailed description of package
+# shortDescription: Concise description of package
+# providerName: Organization/entity providing this package
+# maintainers:
+#   - name: Maintainer 1
+#   - name: Maintainer 2
+```
+We can add these packages to the cluster using `kapp`.
+```bash
+kapp deplot -a simple-app-package -f carvel-artifacts/packages/simple-app.carvel.dev
+```
+(`kubectl apply` would yield the same result)
+
+We should now be able to see our package on the cluster
+```bash
+$ kctrl package list
+Target cluster 'https://127.0.0.1:49841' (nodes: minikube)
+
+Installed packages in namespace 'default'
+
+Name  Package Name           Package Version  Status  
+simp  simple-app.carvel.dev  1.9.0            Reconcile succeeded  
```

Comment:
Is the name getting truncated ? `simp`

Reply:
Looks like I copied output for package install
I created an install called `simp` to test it out
Will change!